execution/executor-types/src/executed_chunk.rs

```diff
@@ -4,149 +4,80 @@
-    pub result_view: ExecutedTrees,
+    pub result_state: StateDelta,
+    pub ledger_info: Option<LedgerInfoWithSignatures>,
     /// If set, this is the new epoch info that should be changed to if this is committed.
     pub next_epoch_state: Option<EpochState>,
-    pub ledger_info: Option<LedgerInfoWithSignatures>,
+    pub ledger_update_output: LedgerUpdateOutput,
 }
 
 impl ExecutedChunk {
-    pub fn new_empty(result_view: ExecutedTrees) -> Self {
-        Self {
-            result_view,
-            ..Default::default()
-        }
-    }
-
     pub fn reconfig_suffix(&self) -> Self {
         assert!(self.next_epoch_state.is_some());
         Self {
-            result_view: self.result_view.clone(),
+            result_state: self.result_state.clone(),
+            ledger_info: None,
             next_epoch_state: self.next_epoch_state.clone(),
-            ..Default::default()
+            ledger_update_output: self.ledger_update_output.reconfig_suffix(),
         }
     }
 
     pub fn transactions_to_commit(&self) -> &Vec<TransactionToCommit> {
-        &self.to_commit
-    }
-
-    pub fn transactions(&self) -> Vec<Transaction> {
-        self.to_commit
-            .iter()
-            .map(|txn_to_commit| txn_to_commit.transaction())
-            .cloned()
-            .collect()
-    }
-
-    pub fn events_to_commit(&self) -> Vec<ContractEvent> {
-        self.to_commit
-            .iter()
-            .flat_map(|txn_to_commit| txn_to_commit.events())
-            .cloned()
-            .collect()
+        &self.ledger_update_output.to_commit
     }
 
     pub fn has_reconfiguration(&self) -> bool {
         self.next_epoch_state.is_some()
     }
 
-    pub fn ensure_transaction_infos_match(
-        &self,
-        transaction_infos: &[TransactionInfo],
-    ) -> Result<()> {
-        ensure!(
-            self.to_commit.len() == transaction_infos.len(),
-            "Lengths don't match. {} vs {}",
-            self.to_commit.len(),
-            transaction_infos.len(),
-        );
-
-        let start_version =
-            self.result_view.txn_accumulator().version() + 1 - self.to_commit.len() as u64;
-        zip_eq(self.to_commit.iter(), transaction_infos.iter())
-            .enumerate()
-            .try_for_each(|(idx, (txn_to_commit, expected_txn_info))| {
-                let txn_info = txn_to_commit.transaction_info();
-                if txn_info != expected_txn_info {
-                    bail!(
-                        "Transaction infos don't match. version: {}, txn_info:{txn_info}, expected_txn_info:{expected_txn_info}",
-                        start_version + idx as u64
-                    );
-                }
-                Ok(())
-            })
+    pub fn combine(&mut self, rhs: Self) {
+        let Self {
+            result_state,
+            ledger_info,
+            next_epoch_state,
+            ledger_update_output,
+        } = rhs;
+        self.result_state = result_state;
```

Comment:
Do you need to combine the HashMap in the result_state?

Reply:
No the "tail" is managed by state_updates_until_latest_checkpoint?